quran/api/views.py

```diff
@@ -0,0 +1,17 @@
+from rest_framework import viewsets
+
+from ..models import Surah
+from .serializers import SurahListSerializer, SurahDetailsSerializer
+
+
+class SurahListRetrieveView(viewsets.ReadOnlyModelViewSet):
+    """View that allows to List, Retrieve Surah."""
+    lookup_field = 'number'
+    queryset = Surah.objects.all()
+    serializer_class = SurahListSerializer
+
+    def get_serializer_class(self):
+        """Returns appreciate serializer class for the given action."""
+        if self.action == "retrieve":
+            return SurahDetailsSerializer
+        return super().get_serializer_class()
```

Comment:
This is a bit confusing. `SurahListSerializer` and `SurahDetailsSerializer` are not entirely symmetric, which makes me think that there is a special logic behind that.
Let's rewrite it without specifying `serializer_class`.

Regarding your comment, I'm not entirely sure, how much value showing override gives. You just have to know that there are special functions. You might indicate it in a comment.